src/neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -199,9 +199,9 @@ public override int GetHashCode()
 
         public UInt160[] GetScriptHashesForVerifying(StoreView snapshot)
         {
-            var hashes = new HashSet<UInt160> { Sender };
-            hashes.UnionWith(Cosigners.Select(p => p.Account));
-            return hashes.OrderBy(p => p).ToArray();
+            var hashes = Cosigners.Select(p => p.Account);
+            hashes = hashes.Contains(Sender) ? hashes : hashes.Append(Sender);
+            return hashes.ToArray();
```

Comment:
The order cannot be removed.

https://github.com/neo-project/neo/blob/3de164622b59f68f617a35f186daf3ac99741a1f/src/neo/SmartContract/Helper.cs#L126-L168

Reply:
But the order of witness is also by func GetScriptHashesForVerifying. So whether we order hashes or not, witnesses and hashes are one-to-one correspondence.

Reply:
You are modifying `GetScriptHashesForVerifying()`, aren't you?

Reply:
https://github.com/neo-project/neo/blob/1451f849d98915775f794684a9dd073cececa6d0/src/neo/SmartContract/ContractParametersContext.cs#L235

and

https://github.com/neo-project/neo/blob/1451f849d98915775f794684a9dd073cececa6d0/src/neo/SmartContract/ContractParametersContext.cs#L93

Actually the order of witness depends on func GetScriptHashesForVerifying, so witnesses and hashes are one-to-one correspondence whether ordered or not.

Reply:
You are talking ablout this implementation. But NEO is a protocol. You need to find a way to ensure the order between different implementations.